core/kernel/dt_driver.c

```diff
@@ -286,6 +289,19 @@ void *dt_driver_device_from_node_idx_prop(const char *prop_name,
 			return NULL;
 		}
 
+		/* Pinctrl states refers to a subnode of the controller node */
```

Comment:
Getting `phandle_node` is usefull only for pinctrl (up to now). `fdt_node_offset_by_phandle()` as well as `fdt_parent_offset()` reparses FDT, better avoid if not usefull.

Reply:
Ok, will move this in the `if (type == DT_DRIVER_PINCTRL)`